radis/lbl/base.py

```diff
@@ -2318,14 +2584,93 @@ def _calc_populations_noneq_multiTvib(
+
+            # TODO: optimize with np.take trick (see equilibrium case)
+
+            for (id, iso), idx in dgb.indices.items():
+
+                molecule = get_molecule(id)
+                state = self.input.state
+                parsum = self.get_partition_function_calculator(molecule, iso, state)
+                Q = parsum.at_noneq_3Tvib(
+                    Tvib,
+                    Trot,
+                    vib_distribution=vib_distribution,
+                    rot_distribution=rot_distribution,
+                    overpopulation=overpopulation,
+                    update_populations=self.misc.export_populations,
+                )
 
-        # TODO: optimize with np.take trick (see equilibrium case)
+                #            df.loc[idx, 'Qvib'] = Qvib
+                df.loc[idx, "Q"] = Q
+                #            dg_list.append(dg)
 
-        for (id, iso), idx in dgb.indices.items():
+                if radis.DEBUG_MODE:
+                    assert (df.loc[idx, "id"] == id).all()
+                    assert (df.loc[idx, "iso"] == iso).all()
 
+        elif "id" not in df and "iso" in df:
+            dgb = df.groupby(by=["iso"])
+            id = df.attrs["id"]
             molecule = get_molecule(id)
+
+            # TODO: optimize with np.take trick (see equilibrium case)
+
+            for (iso), idx in dgb.indices.items():
+
+                state = self.input.state
+                parsum = self.get_partition_function_calculator(molecule, iso, state)
+                Q = parsum.at_noneq_3Tvib(
+                    Tvib,
+                    Trot,
+                    vib_distribution=vib_distribution,
+                    rot_distribution=rot_distribution,
+                    overpopulation=overpopulation,
+                    update_populations=self.misc.export_populations,
+                )
+
+                #            df.loc[idx, 'Qvib'] = Qvib
+                df.loc[idx, "Q"] = Q
+                #            dg_list.append(dg)
+
+                if radis.DEBUG_MODE:
+                    assert (df.loc[idx, "iso"] == iso).all()
+
+        elif "id" in df and "iso" not in df:
+            dgb = df.groupby(by=["id"])
+            iso = df.attrs["iso"]
+
+            # TODO: optimize with np.take trick (see equilibrium case)
+
+            for (id), idx in dgb.indices.items():
+
+                state = self.input.state
+                molecule = get_molecule(id)
+                parsum = self.get_partition_function_calculator(molecule, iso, state)
+                Q = parsum.at_noneq_3Tvib(
+                    Tvib,
+                    Trot,
+                    vib_distribution=vib_distribution,
+                    rot_distribution=rot_distribution,
+                    overpopulation=overpopulation,
+                    update_populations=self.misc.export_populations,
+                )
+
+                #            df.loc[idx, 'Qvib'] = Qvib
+                df.loc[idx, "Q"] = Q
```

Comment:
warning; here and above (L2635) we're re-adding a "Q" column. That's not needed anymore, and not something we want !